crates/sui-core/src/safe_client.rs

```diff
@@ -213,6 +213,47 @@ impl<C, S: SignatureVerifier> SafeClient<C, S> {
         self.verifier.verify_one(signed_effects, &committee)
     }
 
+    fn check_signed_effects_plain(
+        &self,
+        digest: &TransactionDigest,
+        signed_effects: SignedTransactionEffects,
+        expected_effects_digest: Option<&TransactionEffectsDigest>,
+    ) -> SuiResult<SignedTransactionEffects> {
+        // Check it has the right signer
+        fp_ensure!(
+            signed_effects.auth_sig().authority == self.address,
+            SuiError::ByzantineAuthoritySuspicion {
+                authority: self.address,
+                reason: format!(
+                    "Unexpected validator address in the signed effects signature: {:?}",
+                    signed_effects.auth_sig().authority
+                ),
+            }
+        );
+        // Checks it concerns the right tx
+        fp_ensure!(
+            signed_effects.data().transaction_digest() == digest,
+            SuiError::ByzantineAuthoritySuspicion {
+                authority: self.address,
+                reason: "Unexpected tx digest in the signed effects".to_string()
+            }
+        );
+        // check that the effects digest is correct.
+        if let Some(effects_digest) = expected_effects_digest {
+            fp_ensure!(
+                signed_effects.digest() == effects_digest,
+                SuiError::ByzantineAuthoritySuspicion {
+                    authority: self.address,
+                    reason: "Effects digest does not match with expected digest".to_string()
+                }
+            );
+        }
+        match self.get_committee(&signed_effects.epoch()) {
+            Ok(_) => Ok(signed_effects),
+            Err(e) => Err(e),
+        }
+    }
+
     fn check_transaction_info(
```

Comment:
Is there a reason we want to keep the old functions?

Reply:
what does this do? `handle_transaction_info_request_from_some_validators` it still uses check_transaction_info to covert it to the VerifiedTransactionInfoResponse

Reply:
took a closer look, this can be removed, see separate comment